Assets/LeapMotionModules/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,310 @@
+      }
+    }
+
+    // TODO: Do correct thing in OnEnable / OnDisable for whole Managers
+
+    void Start() {
+      if (Provider == null) {
+        Debug.LogError("[InteractionManager] No LeapServiceProvider found.");
+        this.enabled = false;
+      }
+    }
+
+    void FixedUpdate() {
+      OnPrePhysicalUpdate();
+
+      // Ensure provider scale information is up-to-date.
+      if (Provider != null) {
+        _providerScale = Provider.transform.lossyScale.x;
+      }
+
+      // Perform each hand's FixedUpdateHand.
+      foreach (var interactionHand in _interactionHands) {
+        interactionHand.FixedUpdateHand(enableHovering, enableContact, enableGrasping);
+      }
+
+      // Apply soft contacts from both hands in unified solve.
+      // (This will clear softContacts and originalVelocities as well.)
+      if (_softContacts.Count > 0) {
+        PhysicsUtility.applySoftContacts(_softContacts, _softContactOriginalVelocities);
+      }
+
+      // Perform each interaction object's FixedUpdateObject.
+      foreach (var interactionObj in _interactionBehaviours) {
+        interactionObj.FixedUpdateObject();
+      }
+
+      OnPostPhysicalUpdate();
+    }
+
+    void LateUpdate() {
+      OnGraphicalUpdate();
+    }
+
+    #region Object Registration
+
+
+    public void RegisterInteractionBehaviour(InteractionBehaviourBase interactionObj) {
+      _interactionBehaviours.Add(interactionObj);
+      _rigidbodyRegistry[interactionObj.rigidbody] = interactionObj;
+    }
+
+    public bool IsBehaviourRegistered(InteractionBehaviourBase interactionObj) {
+      return _interactionBehaviours.Contains(interactionObj);
+    }
+
+    // TODO: Allow InteractionBehaviours to be unregistered; this should call out to hands and
+    // handle their grasped object state appropriately if their grasped object was just unregistered.
+
+    #endregion
+
+    /// <summary>
+    /// Returns the InteractionHand object that corresponds to the given Hand object.
+    /// 
+    /// Currently, the InteractionManager supports only two InteractionHands at one time: one player's left and right hands.
+    /// </summary>
+    public InteractionHand GetInteractionHand(Hand hand) {
+      if (hand.IsLeft) {
+        return _interactionHands[0];
+      }
+      else {
+        return _interactionHands[1];
+      }
+    }
+
+    /// <summary> Returns true if the object was released from a grasped hand, or false if the object was not held in the first place. </summary>
+    public bool ReleaseObjectFromGrasp(InteractionBehaviourBase interactionObj) {
+      if (!_interactionBehaviours.Contains(interactionObj)) {
+        Debug.LogError("ReleaseObjectFromGrasp was called, but the interaction object " + interactionObj.transform.name + " is not registered"
+          + " with this InteractionManager.");
+        return false;
+      }
+
+      foreach (var hand in _interactionHands) {
+        if (hand.IsGrasping(interactionObj)) {
+          hand.ReleaseGrasp();
```

Comment:
Same as before, when two hands are holding onto an object you don't want to bail on the first hand.